lib/swm050/gpio.c

```diff
@@ -187,11 +183,129 @@ is enabled, GPIO and AF of the SWD pins will be both unavailable.
 */
 void gpio_sel_swd(bool en)
 {
-	if (en) {
-		SWD_SEL = 1;
-	} else {
-		SWD_SEL = 0;
+	SWD_SEL = en;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the pins as external interrupts, rather than normal GPIO
+
+Enable interrupts on the selected pins.  If you want to quickly
+switch on and off interrupts, use gpio_int_mask() after calling this.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_int_en(uint16_t gpios)
+{
+	for (uint8_t i = 0; 0 <= 9; ++i) {
+		if (gpios & (1 << i)) {
+			nvic_enable_irq(NVIC_GPIOA0_IRQ + i);
+		}
 	}
+	INTEN_A |= gpios;
+	gpio_int_mask(gpios, GPIO_UNMASKED);
+}
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets bits in the interrupt mask
+
+When interrupts are masked, it prevents them from being received, which is a
+quicker way to turn on and off GPIO interrupts (after calling gpio_int_en()).
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+	     
+@param[in] gpios Pin mask selection @ref gpio_int_masked
+	     Whether to mask or unmask pins.
+*/
+void gpio_int_mask(uint16_t gpios, bool masked)
+{
+	INTMASK_A = masked ? (INTMASK_A | gpios) : (INTMASK_A & ~gpios);
+}
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets whether the pins are edge triggered or level triggered
+
+Sets whether the pins are edge triggered or level triggered.  Edge-triggered
```

Comment:
as a general doxygen tip, you can drop the @brief, and just remember it uses the first sentence up to the .  Can make things simpler to write and read.